code/modules/projectiles/guns/ballistic/revolver.dm

```diff
@@ -229,6 +228,8 @@
 		user.visible_message("<span class='danger'>*click*</span>")
 		playsound(src, "gun_dry_fire", 30, 1)
 
+	. = ..()
```

Comment:
afterattack needs to call on parent proc at least once in its run so the signal further down can be activated. With this the proc can return without the parent call. You'll need to make some other way of canceling the fire.

Reply:
works fine in testing though? what am i missing here

Reply:
wait are you talking about things that could possibly add more to russian revolver's afterattack, like some sort of attachment? what can i do instead to fix the bug

Reply:
I said, you need to make it possible to cancel the fire while still calling `..()`

Simplest way would be making a check for target further up and nulling target here

Reply:
will see what i can do

Reply:
is it possible to replace all the `return` with `return ..(null, user, flag, params)`?

Reply:
Just changing the arg will change it for `..()`

But yes, what you want to do is possible as well.

Reply:
what if instead of `. = ..()` at the bottom, it was replaced with `. = ..(null, user, flag, params)` instead?

wait just saw in my email you responded to what i said before but deleted, ill try that though

Reply:
The proc still returns earlier on, when that happens it would prevent the call

Reply:
meant to say `. = ..(null, user, flag, params)` would be at the top

Reply:
I'll leave you to figure this out, get on the irc if you want actual help with this. Github isn't the best place for this.